docs/root/start/sandboxes/double-proxy.rst

```diff
@@ -0,0 +1,174 @@
+certificates for the proxies.
+
+.. include:: _include/docker-env-setup.rst
+
+Change to the ``examples/double-proxy`` directory.
+
+Step 3: Create a certificate authority
+**************************************
+
+First create a key for the certificate authority:
+
+.. code-block:: console
+
+   $ pwd
+   envoy/examples/double-proxy
+   $ mkdir -p certs
+   $ openssl genrsa -out certs/ca.key 4096
+   Generating RSA private key, 4096 bit long modulus (2 primes)
+   ..........++++
+   ..........................................................................................................++++
+   e is 65537 (0x010001)
+
+Now use the key to generate a certificate authority certificate.
+
+If you wish, you can interactively alter the fields in the certificate.
+
+For the purpose of this example, the defaults should be sufficient.
+
+.. code-block:: console
+
+   $ openssl req -x509 -new -nodes -key certs/ca.key -sha256 -days 1024 -out certs/ca.crt
+
+   You are about to be asked to enter information that will be incorporated
+   into your certificate request.
+   What you are about to enter is what is called a Distinguished Name or a DN.
+   There are quite a few fields but you can leave some blank
+   For some fields there will be a default value,
+   If you enter '.', the field will be left blank.
+   -----
+   Country Name (2 letter code) [AU]:
+   State or Province Name (full name) [Some-State]:
+   Locality Name (eg, city) []:
+   Organization Name (eg, company) [Internet Widgits Pty Ltd]:
+   Organizational Unit Name (eg, section) []:
+   Common Name (e.g. server FQDN or YOUR name) []:
+   Email Address []:
+
+Step 4: Create a domain key
+***************************
+
+Create a key for the example domain:
+
+.. code-block:: console
+
+   $ openssl genrsa -out certs/example.com.key 2048
+   Generating RSA private key, 2048 bit long modulus (2 primes)
+   ..+++++
+   .................................................+++++
+   e is 65537 (0x010001)
+
+Step 5: Generate certificate signing requests for the proxies
+*************************************************************
+
+Use the domain key to generate certificate signing requests for each of the proxies:
+
+.. code-block:: console
+
+   $ openssl req -new -sha256 \
+        -key certs/example.com.key \
+        -subj "/C=US/ST=CA/O=MyExample, Inc./CN=proxy-postgres-frontend.example.com" \
+        -out certs/proxy-postgres-frontend.example.com.csr
+   $ openssl req -new -sha256 \
+        -key certs/example.com.key \
+        -subj "/C=US/ST=CA/O=MyExample, Inc./CN=proxy-postgres-backend.example.com" \
+        -out certs/proxy-postgres-backend.example.com.csr
+
+Step 6: Sign the proxy certificates
+***********************************
+
+You can now use the certificate authority that you created to sign the certificate requests.
+
+Note the ``subjectAltName``. This is used for reciprocally matching and validating the certificates.
```

Comment:
afaict this is what is matched in sni but i maybe wrong (im not familiar with exactly how sni works)